### ISSUE TITLE
`Chain.capture()` fails to generate complete stack trace for `FileSystemException`

### ISSUE THREAD

**Issue by cbatson** [dart-lang/stack_trace#124]
## Steps to reproduce

Run the following code:

```dart
import 'dart:io';
import 'package:stack_trace/stack_trace.dart';

Future<void> badFunc() => File('foo').readAsString();

void main(List<String> arguments) async {
  await Chain.capture(() async {
    try {
      await badFunc();
    }
    catch (error, stackTrace) {
      print(stackTrace);
    }
  });
}
```

## Output

```
#0      _File.open.<anonymous closure> (dart:io/file_impl.dart:356:9)
#1      StackZoneSpecification._registerUnaryCallback.<anonymous closure>.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:124:36)
#2      StackZoneSpecification._run (package:stack_trace/src/stack_zone_specification.dart:204:15)
#3      StackZoneSpecification._registerUnaryCallback.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:124:24)
#4      _rootRunUnary (dart:async/zone.dart:1399:47)
#5      _CustomZone.runUnary (dart:async/zone.dart:1300:19)
<asynchronous suspension>
#6      StackZoneSpecification._registerUnaryCallback.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:124:15)
<asynchronous suspension>
```

## Expected result

I am expecting to see some mention of `main()` and/or the source file name in the stack trace. However, as you can see, there is **no** mention of any user code in this stack trace at all.

Using `stack_trace` version 1.11.0.

**Comment by blaugold** [comment]
@mraleph Is this issue going to be fixed by d3e4c4d633eca379462dd4473b8995761155ee84?

**Comment by mraleph** [comment]
@blaugold yep. This is now fixed (if you use 1.11.1 and new enough Dart SDK):

```
#0      _checkForErrorResponse (dart:io/common.dart:55:9)
#1      _File.open.<anonymous closure> (dart:io/file_impl.dart:381:7)
#2      StackZoneSpecification._registerUnaryCallback.<anonymous closure>.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:127:36)
#3      StackZoneSpecification._run (package:stack_trace/src/stack_zone_specification.dart:207:15)
#4      StackZoneSpecification._registerUnaryCallback.<anonymous closure> (package:stack_trace/src/stack_zone_specification.dart:127:24)
#5      _rootRunUnary (dart:async/zone.dart:1407:47)
#6      _CustomZone.runUnary (dart:async/zone.dart:1308:19)
<asynchronous suspension>
#7      _File.readAsBytes.<anonymous closure> (dart:io/file_impl.dart:562:24)
<asynchronous suspension>
#8      _File.readAsString (dart:io/file_impl.dart:621:18)
<asynchronous suspension>
#9      main.<anonymous closure> (file:///Users/vegorov/src/temp/test_stack_trace/bin/main.dart:9:7)
<asynchronous suspension>
#10     main (file:///Users/vegorov/src/temp/test_stack_trace/bin/main.dart:7:3)
<asynchronous suspension>
```
